### PR TITLE
[xdoctest] reformat example code with google style No.116-119

### PR DESCRIPTION
**Pull request by AndSonder**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码为新的格式，并通过 xdoctest 检查：

1. python/paddle/optimizer/optimizer.py
2. python/paddle/quantization/config.py 
3. python/paddle/quantization/factory.py
4. python/paddle/quantization/imperative/qat.py

- #55629

### CODE REVIEW

File: python/paddle/optimizer/optimizer.py [hunk 1]
```diff
@@ -500,23 +500,23 @@ def set_lr(self, value):
...
+
+                >>> adam = paddle.optimizer.Adam(0.1, parameters=linear.parameters())
+
+                >>> # set learning rate manually by python float value
+                >>> lr_list = [0.2, 0.3, 0.4, 0.5, 0.6]
+                >>> for i in range(5):
+                ...     adam.set_lr(lr_list[i])
+                ...     lr = adam.get_lr()
+                ...     print("current lr is {}".format(lr))
+                Print:
```
Comment (SigureMo) [inline python/paddle/optimizer/optimizer.py]: image

输出结果里没有 `Print:`

File: python/paddle/quantization/imperative/qat.py [hunk 2]
```diff
@@ -138,76 +138,76 @@ def __init__(
...
-            # Fine-tune the quantized model
-            # ...
-
-            # Save quant model for the inference.
-            imperative_qat.save_quantized_model(
-                layer=model,
-                model_path="./imperative_model_qat")
+            >>> import paddle
+            >>> from paddle.static.quantization \
+            ...     import ImperativeQuantAware
```
Comment (SigureMo) [inline python/paddle/quantization/imperative/qat.py]: import 代码风格改一下吧

```python
>>> from paddle.static.quantization import (
...     ImperativeQuantAware,
... )
```

File: python/paddle/quantization/config.py [hunk 3]
```diff
@@ -70,12 +70,12 @@ class QuantConfig:
...
-          from paddle.quantization.quanters import FakeQuanterWithAbsMaxObserver
+          >>> from paddle.quantization import QuantConfig
+          >>> from paddle.quantization.quanters import FakeQuanterWithAbsMaxObserver
 
-          quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
-          q_config = QuantConfig(activation=quanter, weight=quanter)
-          print(q_config)
+          >>> quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
+          >>> q_config = QuantConfig(activation=quanter, weight=quanter)
+          >>> print(q_config)
```
Comment (SigureMo) [inline python/paddle/quantization/config.py:78]: 这里输出方便打印么？方便的话最好加一下输出，如果输出随机可以考虑 skip，如果输出太多可以考虑用 `...` 或者不写

**Comment by AndSonder** [comment]
@SigureMo  Done~

File: python/paddle/quantization/config.py [hunk 4]
```diff
@@ -111,20 +114,20 @@ def add_layer_config(
...
+
+             >>> class Model(paddle.nn.Layer):
+             ...    def __init__(self):
+             ...        super().__init__()
+             ...        self.fc = Linear(576, 120)
+             >>> model = Model()
+             >>> quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
+             >>> q_config = QuantConfig(activation=None, weight=None)
+             >>> q_config.add_layer_config([model.fc], activation=quanter, weight=quanter)
+             >>> print(q_config)
```
Comment (SigureMo) [inline python/paddle/quantization/config.py]: 这里

File: python/paddle/quantization/config.py [hunk 5]
```diff
@@ -155,20 +158,20 @@ def add_name_config(
...
+
+             >>> class Model(paddle.nn.Layer):
+             ...     def __init__(self):
+             ...         super().__init__()
+             ...         self.fc = Linear(576, 120)
+             >>> model = Model()
+             >>> quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
+             >>> q_config = QuantConfig(activation=None, weight=None)
+             >>> q_config.add_name_config([model.fc.full_name()], activation=quanter, weight=quanter)
+             >>> print(q_config)
```
Comment (SigureMo) [inline python/paddle/quantization/config.py]: 还有这里

File: python/paddle/quantization/config.py [hunk 6]
```diff
@@ -200,20 +203,20 @@ def add_type_config(
...
+
+            >>> class Model(paddle.nn.Layer):
+            ...     def __init__(self):
+            ...         super().__init__()
+            ...         self.fc = Linear(576, 120)
+            >>> model = Model()
+            >>> quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
+            >>> q_config = QuantConfig(activation=None, weight=None)
+            >>> q_config.add_type_config([Linear], activation=quanter, weight=quanter)
+            >>> print(q_config)
```
Comment (SigureMo) [inline python/paddle/quantization/config.py]: 和这里

**Review by SigureMo — DISMISSED** [review]
都一样哈，如果合适就贴一下输出，不合适就不用了

File: python/paddle/quantization/config.py (hunk 6 above)
Comment (AndSonder) [reply python/paddle/quantization/config.py]: done

File: python/paddle/quantization/config.py [hunk 7]
```diff
@@ -111,20 +114,24 @@ def add_layer_config(
...
+             ...        self.fc = Linear(576, 120)
+             >>> model = Model()
+             >>> quanter = FakeQuanterWithAbsMaxObserver(moving_rate=0.9)
+             >>> q_config = QuantConfig(activation=None, weight=None)
+             >>> q_config.add_layer_config([model.fc], activation=quanter, weight=quanter)
+             >>> print(q_config)
+             Global config:
+             None
+             Layer prefix config: 
+             {'linear_0': <paddle.quantization.config.SingleLayerConfig object at 0x7fe41a680ee0>}
```
Comment (SigureMo) [inline python/paddle/quantization/config.py]: 这里有内存地址的肯定不能保证输出一致的，可以用 `# doctest: +SKIP` 跳过输出检查

Comment (AndSonder) [reply python/paddle/quantization/config.py]: > 这里有内存地址的肯定不能保证输出一致的，可以用 `# doctest: +SKIP` 跳过输出检查

done

**Comment by SigureMo** [comment]
code style 需要处理一下

**Comment by AndSonder** [comment]
> code style 需要处理一下

前面说的import的style已经都处理完了

**Comment by SigureMo** [comment]
image

codestyle 流水线，应该是有 trailing whitespace，pre-commit 处理一下就行了

File: python/paddle/quantization/factory.py [hunk 8]
```diff
@@ -83,21 +83,21 @@ def quanter(class_name):
...
+            ...     def __init__(self, arg1, kwarg1=None):
+            ...         pass
+
+            >>> # Used in ./test.py
+            >>> # from .customized_quanter import CustomizedQuanter
+            >>> from paddle.quantization import QuantConfig
+            >>> arg1_value = "test"
+            >>> kwarg1_value = 20
+            >>> quanter = CustomizedQuanter(arg1_value, kwarg1=kwarg1_value)
+            >>> q_config = QuantConfig(activation=quanter, weight=quanter)
```
Comment (SigureMo) [inline python/paddle/quantization/factory.py:102]: 这段代码看起来不能在 xdoctest 环境跑，skip 掉吧

**Review by SigureMo — DISMISSED** [review]
LGTMeow 🐾

File: python/paddle/quantization/config.py [hunk 9]
```diff
@@ -155,20 +163,25 @@ def add_name_config(
          Examples:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/quantization/config.py]: .. code-block::python 要缩进

File: python/paddle/quantization/config.py [hunk 10]
```diff
@@ -200,20 +213,25 @@ def add_type_config(
         Examples:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/quantization/config.py]: 要缩进

File: python/paddle/quantization/config.py [hunk 11]
```diff
@@ -242,16 +260,16 @@ def add_qat_layer_mapping(self, source: type, target: type):
...
```
Comment (megemini) [inline python/paddle/quantization/config.py]: 要缩进

File: python/paddle/quantization/config.py [hunk 12]
```diff
@@ -274,11 +292,11 @@ def add_customized_leaf(self, layer_type: type):
...
```
Comment (megemini) [inline python/paddle/quantization/config.py]: 要缩进

File: python/paddle/quantization/config.py [hunk 13]
```diff
@@ -381,20 +399,20 @@ def _specify(self, model: Layer):
...
```
Comment (megemini) [inline python/paddle/quantization/config.py]: 要缩进

File: python/paddle/quantization/imperative/qat.py [hunk 14]
```diff
@@ -138,76 +138,79 @@ def __init__(
         Examples 1:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/quantization/imperative/qat.py]: 要缩进

File: python/paddle/quantization/imperative/qat.py [hunk 15]
```diff
@@ -138,76 +138,79 @@ def __init__(
         Examples 1:
```
Comment (megemini) [inline python/paddle/quantization/imperative/qat.py]: 不能写 `Examples 1:`，需要改为 

``` python
Examples:

    .. code-block:: python
       :name: xxx(比如code-example-xxx)

        >>> xxx
```

File: python/paddle/quantization/imperative/qat.py [hunk 16]
```diff
@@ -138,76 +138,79 @@ def __init__(
...
+
+            >>> # Save quant model for the inference.
+            >>> imperative_qat.save_quantized_model(
+            ...     layer=model,
+            ...     model_path="./resnet50_qat",
+            ...     input_spec=[
+            ...         paddle.static.InputSpec(
+            ...         shape=[None, 3, 224, 224], dtype='float32')])
 
         Examples 2:
```
Comment (megemini) [inline python/paddle/quantization/imperative/qat.py]: 同上～ 同时注意缩进

File: python/paddle/quantization/imperative/qat.py [hunk 17]
```diff
@@ -247,37 +250,38 @@ def quantize(self, model):
         Examples:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/quantization/imperative/qat.py]: 要缩进

File: python/paddle/optimizer/optimizer.py [hunk 18]
```diff
@@ -611,49 +610,49 @@ def get_lr(self):
...
+                >>> import numpy as np
+                >>> emb = paddle.nn.Embedding(10, 3)
+
+                >>> ## example1: LRScheduler is not used, return the same value is all the same
+                >>> adam = paddle.optimizer.Adam(0.01, parameters = emb.parameters())
+                >>> for batch in range(10):
+                ...     input = paddle.randint(low=0, high=5, shape=[5])
+                ...     out = emb(input)
+                ...     out.backward()
+                ...     print("Learning rate of step{}: {}".format(batch, adam.get_lr())) # 0.01
```
Comment (megemini) [inline python/paddle/optimizer/optimizer.py:624]: 这里要有输出～

File: python/paddle/optimizer/optimizer.py [hunk 19]
```diff
@@ -611,49 +610,49 @@ def get_lr(self):
...
+                >>> ## example2: StepDecay is used, return the scheduled learning rate
+                >>> scheduler = paddle.optimizer.lr.StepDecay(learning_rate=0.5, step_size=2, gamma=0.1)
+                >>> adam = paddle.optimizer.Adam(scheduler, parameters = emb.parameters())
+                >>> for batch in range(10):
+                ...     input = paddle.randint(low=0, high=5, shape=[5])
+                ...     out = emb(input)
+                ...     out.backward()
+                ...     print("Learning rate of step{}: {}".format(batch, adam.get_lr())) # 0.5->0.05...
+                ...     adam.step()
+                ...     scheduler.step()
```
Comment (megemini) [inline python/paddle/optimizer/optimizer.py:646]: 这里要有输出～

File: python/paddle/optimizer/optimizer.py [hunk 20]
```diff
@@ -611,49 +610,49 @@ def get_lr(self):
...
+                ...     scheduler = paddle.optimizer.lr.StepDecay(learning_rate=0.5, step_size=2, gamma=0.1)
+                ...     adam = paddle.optimizer.Adam(learning_rate=scheduler)
+                ...     adam.minimize(loss)
+
+                >>> exe = paddle.static.Executor()
+                >>> exe.run(start_prog)
+                >>> for batch in range(10):
+                ...     print("Learning rate of step{}: {}", adam.get_lr())     # 0.5->0.05->0.005...
+                ...     out = exe.run(main_prog, feed={'x': np.random.randn(3, 10).astype('float32')})
+                ...     scheduler.step()
```
Comment (megemini) [inline python/paddle/optimizer/optimizer.py]: 要有输出～

File: python/paddle/quantization/factory.py [hunk 21]
```diff
@@ -83,21 +83,22 @@ def quanter(class_name):
...
+            ...         pass
+
+            >>> # Used in ./test.py
+            >>> # from .customized_quanter import CustomizedQuanter
+            >>> from paddle.quantization import QuantConfig
+            >>> # doctest: +SKIP
+            >>> arg1_value = "test"
+            >>> kwarg1_value = 20
+            >>> quanter = CustomizedQuanter(arg1_value, kwarg1=kwarg1_value)
+            >>> q_config = QuantConfig(activation=quanter, weight=quanter)
```
Comment (megemini) [inline python/paddle/quantization/factory.py]: 整段 skip 吧～

**Comment by AndSonder** [comment]
麻烦再review一下~ @megemini

File: python/paddle/optimizer/optimizer.py [hunk 22]
```diff
@@ -611,50 +610,79 @@ def get_lr(self):
...
+                Learning rate of step{}: {} 0.5
+                Learning rate of step{}: {} 0.5
+                Learning rate of step{}: {} 0.05
+                Learning rate of step{}: {} 0.05
+                Learning rate of step{}: {} 0.005000000000000001
+                Learning rate of step{}: {} 0.005000000000000001
+                Learning rate of step{}: {} 0.0005000000000000001
+                Learning rate of step{}: {} 0.0005000000000000001
+                Learning rate of step{}: {} 5.000000000000001e-05
+                Learning rate of step{}: {} 5.000000000000001e-05
```
Comment (megemini) [inline python/paddle/optimizer/optimizer.py]: print("Learning rate of step{}: {}", adam.get_lr()) -> print("Learning rate of step{}: {}", batch, adam.get_lr())

File: python/paddle/quantization/config.py [hunk 23]
```diff
@@ -111,20 +114,25 @@ def add_layer_config(
          Examples:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/quantization/config.py]: .. code-block:: python 要有缩进～

**Review by megemini — COMMENTED** [review]
`Examples 1:` 要改一下哈～

File: python/paddle/quantization/config.py (hunk 23 above)
Comment (SigureMo) [reply python/paddle/quantization/config.py]: 这居然还反向缩进了:joy:，缩退么 [doge]

整段 docstring 的缩进都有问题，不要多余的一个空格

Comment (AndSonder) [reply python/paddle/quantization/config.py]: > 

这个是原来的缩进就有问题，我改漏了, 已经改过来了, 实在不好意思，改的时候可能整段的缩进出问题了

File: python/paddle/quantization/imperative/qat.py [hunk 24]
```diff
@@ -135,79 +135,82 @@ def __init__(
...
+                >>> # fake quant logical.
+                >>> imperative_qat.quantize(model)
+
+                >>> # Fine-tune the quantized model
+                >>> # ...
+
+                >>> # Save quant model for the inference.
+                >>> imperative_qat.save_quantized_model(
+                ...    layer=model,
+                ...    model_path="./imperative_model_qat")
```
Comment (SigureMo) [inline python/paddle/quantization/imperative/qat.py:212]: 不要写成

```
Examples:
    .. code-block:: python

        >>> xxxxx

Examples:
    .. code-block:: python

        >>> yyyyy
```

而是写成

```
Examples:
    .. code-block:: python
        :name: code-example-1

        >>> xxxxx

    .. code-block:: python
        :name: code-example-2

        >>> yyyyy
```

因为本 API 非公开 API，所以就不用添加 name 了，写成下面即可：

```
Examples:
    .. code-block:: python

        >>> xxxxx

    .. code-block:: python

        >>> yyyyy
```

简单来说就是删除第二个 `Examples:` 这一行

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾
